core/Pos/Binary/Class/Core.hs

```diff
@@ -68,6 +70,16 @@ matchSize requestedSize lbl actualSize =
   when (actualSize /= requestedSize) $
     fail (lbl <> " failed the size check. Expected " <> show requestedSize <> ", found " <> show actualSize)
 
+-- | Tries to decode Tag 24 (CBOR Data Item) from a ByteString blob,
```

Comment:
Why not use `decodeTag` explicitly? Here's the implementation I proposed in https://issues.serokell.io/issue/CSL-1494:

```
newtype Cbor = Cbor ByteString

instance Bi Cbor where
    encode (Cbor x) = encodeTag 24 <> encode x
    decode = do
        t <- decodeTag
        when (t /= 24) $ fail $
            "decode@Cbor: expected a bytestring with \
            \CBOR (marked by tag 24), found tag: " <> show t
        decode @ByteString
```